### PR TITLE
Pract.01: Add 07-08| Gabriela Mladenova

### PR DESCRIPTION
**Pull request by gabrielavm**
Gabriela Mladenova
SI 1 3MI0600225

### CODE REVIEW

File: Seminars/Sem.01/Pract.01/Task07.cpp [hunk 1]
```diff
@@ -0,0 +1,110 @@
+//Gabriela Mladenova
+//SI
+//Group 1
+//FN: 3MI0600225
+#include <iostream>
+
+const int SIZEOFARRAYS = 30;
+enum MovieGenre
```
Comment (khatoni) [inline Seminars/Sem.01/Pract.01/Task07.cpp]: Use enum class

File: Seminars/Sem.01/Pract.01/Task07.cpp [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    Horror,
+    Romantic
+};
+struct Movie
+{
+    char movieName[SIZEOFARRAYS + 1];
+    int duration;
+    MovieGenre movieGenre;
+    double rating;
+};
```
Comment (khatoni) [inline Seminars/Sem.01/Pract.01/Task07.cpp:23]: Leave empty lines after structures and functions

File: Seminars/Sem.01/Pract.01/Task07.cpp [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+#include <iostream>
+
+const int SIZEOFARRAYS = 30;
+enum MovieGenre
+{
+    Comedy,
+    Action,
+    Horror,
+    Romantic
+};
```
Comment (khatoni) [inline Seminars/Sem.01/Pract.01/Task07.cpp:15]: Leave empty line

File: Seminars/Sem.01/Pract.01/Task07.cpp [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        }
+        else if (MovieGenre::Romantic)
+        {
+            std::cout << "Romantic" << std::endl;
+        }
+        else
+        {
+            std::cout << "No genre" << std::endl;
+        }
+
```
Comment (khatoni) [inline Seminars/Sem.01/Pract.01/Task07.cpp]: It would be better to use switch case

File: Seminars/Sem.01/Pract.01/Task07.cpp [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+        std::cout << "Movie rating: " << movies[i].rating << std::endl;
+        std::cout << std::endl;
+    }
+}
```
Comment (khatoni) [inline Seminars/Sem.01/Pract.01/Task07.cpp:55]: empty line

File: Seminars/Sem.01/Pract.01/Task07.cpp [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        }
+    }
+    for (int j = 0; j < SIZE; ++j)
+    {
+        if (j == topMovieIndex)
+        {
+            std::cout << movies[j].movieName << " " << movies[j].duration
+                << " " << movies[j].movieGenre << " " << movies[j].rating;
+        }
+    }
```
Comment (khatoni) [inline Seminars/Sem.01/Pract.01/Task07.cpp]: Just write cout<<movies[topMovieIndex].movieName<<" "<<movies[topMovieIndex].duration and so on.
This for loop is useless

File: Seminars/Sem.01/Pract.01/Task07.cpp [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+}
```
Comment (khatoni) [inline Seminars/Sem.01/Pract.01/Task07.cpp:71]: empty line

File: Seminars/Sem.01/Pract.01/Task07.cpp [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+void swapMovies(Movie movies[], const int firstIndex, const int secondIndex)
+{
+    Movie temp = movies[firstIndex];
+    movies[firstIndex] = movies[secondIndex];
+    movies[secondIndex] = temp;
+}
```
Comment (khatoni) [inline Seminars/Sem.01/Pract.01/Task07.cpp:78]: empty line

File: Seminars/Sem.01/Pract.01/Task07.cpp [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+{
+    for (int i = 0; i < SIZE - 1; ++i)
+    {
+        for (int j = 0; j < SIZE - i - 1; ++j)
+        {
+            if (movies[j].rating < movies[j + 1].rating)
+            {
+                swapMovies(movies, j, j + 1);
+            }
+        }
```
Comment (khatoni) [inline Seminars/Sem.01/Pract.01/Task07.cpp]: And can you think of algorithm which will be better in this situation.

File: Seminars/Sem.01/Pract.01/Task08.cpp [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+    std::cout << "Enter the number of students in the group: ";
+    std::cin >> group.numberOfStudents;
+
+    for (int i = 0; i < group.numberOfStudents; ++i)
+    {
+        Student student;
+        inputStudentsInfo(student);
+        group.studentsList[i] = student;
+        sumOfGrades += student.grade;
+        ++counter;
```
Comment (khatoni) [inline Seminars/Sem.01/Pract.01/Task08.cpp]: inputStudentInfo(group.studentList[i]);

File: Seminars/Sem.01/Pract.01/Task08.cpp [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+    }
+    group.averageGrade = sumOfGrades / counter;
```
Comment (khatoni) [inline Seminars/Sem.01/Pract.01/Task08.cpp]: This counter is useless, you can use numberOfStudents

File: Seminars/Sem.01/Pract.01/Task08.cpp [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+}
```
Comment (khatoni) [inline Seminars/Sem.01/Pract.01/Task08.cpp:46]: empty line

File: Seminars/Sem.01/Pract.01/Task08.cpp [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+int countOfScholarships(const Group& group, const double minGrade)
+{
+    int counter = 0;
+    for (int i = 0; i < group.numberOfStudents; ++i)
+    {
+        if (group.studentsList[i].grade >= minGrade)
+        {
+            ++counter;
+        }
+    }
```
Comment (khatoni) [inline Seminars/Sem.01/Pract.01/Task08.cpp:58]: empty line after loops

File: Seminars/Sem.01/Pract.01/Task08.cpp [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+    return counter;
+}
```
Comment (khatoni) [inline Seminars/Sem.01/Pract.01/Task08.cpp:61]: empty line

File: Seminars/Sem.01/Pract.01/Task08.cpp [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+void printScholarshipsList(const Group& group, const double minGrade, Student* student)
```
Comment (khatoni) [inline Seminars/Sem.01/Pract.01/Task08.cpp]: What is Student* student?

**Review by khatoni — CHANGES_REQUESTED** [review]

**Comment by Justsvetoslavov** [comment]
Good solutions. Good job 👍

**Pull request closed without merging**